services/network/client/application/tcp-check.usecase.go

```diff
@@ -0,0 +1,86 @@
+/* Copyright (c) 2023 Snowflake Inc. All rights reserved.
+
+   Licensed under the Apache License, Version 2.0 (the
+   "License"); you may not use this file except in compliance
+   with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing,
+   software distributed under the License is distributed on an
+   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+   KIND, either express or implied.  See the License for the
+   specific language governing permissions and limitations
+   under the License.
+*/
+
+package application
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	pb "github.com/Snowflake-Labs/sansshell/services/network"
+)
+
+type TCPCheckResult struct {
```

Comment:
it makes app hardly depends on chosen grpc generation library. Any significant changes in library or proto files will require a lot of changes. If we use own structures we could cover usecase with unit tests and be sure that cli controller would work if it works

Reply:
> it makes app hardly depends on chosen grpc generation library. Any significant changes in library or proto files will require a lot of changes. If we use own structures we could cover usecase with unit tests and be sure that cli controller would work if it works

Why would depending on the grpc generation be a problem? The grpc libraries have interface stability guarantees and the protos are part of SansShell's public API and need to keep backwards compatibility as part of that API.

Reply:
Dependency on any transport specific types in app/business logic force you to change app logic for it. E.g. 
- to storing `port` in go it enough uint8 and app/domain logic (unfortunately we not have it) could use this type, but protobuf not support it and I forced to use uint32. If we look into future when we would extend usecase it would force us to convert port from pb to uint8 and vise versa to use it
- This point would go even further. I'm creating GRPC's `TCPCheckRequest` in app logic and need to convert port to uint32 without any business reason. I mix real app logic and logic what appear only because of GRPC and it makes harder to understand what and why is happening. To avoid that we need to create some adapter to wrap GRPC client and introduce our own types what would be 100% suitable for business logic without unnecessary conversion

I used such approaches before and saw good results. For this particular project I'm not sure yet. I would trust you experience until I had strong enough arguments

Reply:
I'd rather keep the existing conventions of the codebase for now. I'm open to revisiting later on.

If we wanted to change it, we should have a solid plan for changing it throughout the codebase so that we don't end up with different styles of abstraction based on when the code was written.

(My own coding style tends to avoid abstractions and indirections until I have 2-3 concrete use cases for them. Once the use cases emerge, I refactor the code to add in the abstractions.)